python/cudf/cudf/core/dataframe.py

```diff
@@ -3726,6 +3726,112 @@ def sort_values(
+                raise AttributeError(f"{aggs} is not a valid function for 'DataFrame' object")
+            result = cudf.DataFrame()
+            result[aggs] = getattr(df_normalized,aggs)()
+            result = result.T.loc[aggs]
+            result.name = None
+            return result
+
+        elif isinstance(aggs, dict):
+            cols=aggs.keys()
+            if any([callable(val) for val in aggs.values()]):
+                raise NotImplementedError("callable parameter is not implemented yet")
+            elif all([isinstance(val, str) for val in aggs.values()]):
+                result = cudf.Series(index=cols)
+                for key, value in aggs.items():
+                    col = df_normalized[key]
+                    if not hasattr(col, value):
+                        raise AttributeError(f"{value} is not a valid function for 'Series' object")
+                    result[key]= getattr(col, value)()
+            elif all([isinstance(val, Iterable) for val in aggs.values()]):
+                idxs=set()
+                for val in aggs.values():
+                    if isinstance(val, Iterable):
+                        idxs.update(val)
+                    elif isinstance(val, str):
+                        idxs.add(val)    
+                idxs = sorted(list(idxs))
+                for agg in idxs:
+                    if agg is callable:
+                        raise NotImplementedError("callable parameter is not implemented yet")
+                result = cudf.DataFrame(index=idxs, columns=cols)
+                for key in aggs.keys():
+                    col = df_normalized[key]
+                    col_empty =  column_empty(len(idxs), dtype=col.dtype, masked=True)
+                    ans = cudf.Series(data=col_empty, index=idxs)
+                    if isinstance(aggs.get(key), Iterable):
+                        # TODO : Allow simultaneous pass for multi-aggregation as a future optimization
+                        for agg in aggs.get(key):
+                            if not hasattr(col, agg):
+                                raise AttributeError(f"{agg} is not a valid function for 'Series' object")
+                            ans[agg] = getattr(col, agg)()
+                    elif isinstance(aggs.get(key),str):
+                        if not hasattr(col, aggs.get(key)):
+                                raise AttributeError(f"{aggs.get(key)} is not a valid function for 'Series' object")
+                        ans[aggs.get(key)] = getattr(col, agg)()
+                    result[key] = ans
+
+            else: raise ValueError("values of dict must be a string or list")
+
+            return result 
+                
+        elif callable(aggs):
+            raise NotImplementedError("callable parameter is not implemented yet")
+
+        else:
+            raise ValueError("argument must be a string, list or dict")
+
+        return result.T.sort_index(axis=1, ascending=True)
```

Comment:
It looks like this `return` statement only takes care of this case:
```python
if isinstance(aggs, Iterable) and not isinstance(aggs, (str, dict)):
```
Perhaps we should move it into that block?